server/webdriver_tests/screenshot/screenshot_test.py

```diff
@@ -23,77 +25,42 @@
-        'selector':
-            By.CLASS_NAME,
-        'name':
-            'chart-area',
-        'height':
-            1000
-    },
-    {
-        'url': '/ranking/Median_Income_Person/County/country/USA',
-        'selector': By.CLASS_NAME,
-        'name': 'chart-container',
-        'height': 1080
-    },
-    {
-        'url':
-            '/tools/map#%26sv%3DAnnual_Emissions_CarbonDioxide_NonBiogenic%26pc%3D0%26denom%3DCount_Person%26pd%3Dcountry%2FUSA%26ept%3DState%26ppt%3DEpaReportingFacility',
-        'selector':
-            By.ID,
-        'name':
-            'map-items',
-        'height':
-            1080
-    },
-]
+CHART_HOLDER_CLASS = 'dc-chart-holder'
+CHART_EXIST_CLASS = 'dc-chart-exist'
+
+
+def wait_for_charts(driver):
+  element_present = EC.presence_of_element_located(
+      (By.CLASS_NAME, CHART_HOLDER_CLASS))
+  WebDriverWait(driver, 30).until(element_present)
+  chart_containers = driver.find_elements(By.CLASS_NAME, CHART_HOLDER_CLASS)
+  for c in chart_containers:
+    try:
+      c.find_element(By.CLASS_NAME, CHART_EXIST_CLASS)
+    except TimeoutException:
+      return False
+  return True
 
 
 # Class to test timeline tool.
 class TestScreenShot(WebdriverBaseTest):
 
   def test_pages_and_sreenshot(self):
     """Test these page can show correctly and do screenshot."""
-    for _, test_info in enumerate(TEST_URLS):
-      self.driver.get(self.url_ + test_info['url'])
-
-      name = urllib.parse.quote_plus(test_info['url'].removeprefix('/'))
-
-      # Wait until the test_class_name has loaded.
-      element_present = EC.presence_of_element_located(
-          (test_info['selector'], test_info['name']))
-
-      try:
-        WebDriverWait(self.driver, self.TIMEOUT_SEC).until(element_present)
-      except Exception as e:
-        logging.error(test_info['url'])
-        raise e
-
-      # Set the window size. Testing different sizes.
-      self.driver.set_window_size(width=WIDTH,
-                                  height=test_info['height'],
-                                  windowHandle='current')
-
-      # Get the element to test.
-      charts = self.driver.find_elements(test_info['selector'],
-                                         test_info['name'])
-
-      # Assert there is at least one chart.
-      self.assertGreater(len(charts), 0, test_info['url'])
+    with open(
+        os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                     'test_page.json')) as f:
+      data = json.load(f)
+      for test_info in data:
+        self.driver.get(self.url_ + test_info['url'])
 
-      # Take a screenshot of the page and save it.
-      file_name = '{}/{}.png'.format(SCREENSHOTS_FOLDER, name)
-      self.assertTrue(self.driver.save_screenshot(file_name))
+        name = urllib.parse.quote_plus(test_info['url'].removeprefix('/'))
+        # Set the window size. Testing different sizes.
+        self.driver.set_window_size(width=WIDTH,
```

Comment:
wonder if we should be testing different widths too because I think widths make a bigger difference on the charts than height